browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc

```diff
@@ -203,6 +213,42 @@ void AIChatUIPageHandler::GetAPIResponseError(
   std::move(callback).Run(active_chat_tab_helper_->GetCurrentAPIError());
 }
 
+void AIChatUIPageHandler::LikeMessage(bool is_liked,
+                                      const std::string& turn_text,
+                                      LikeMessageCallback callback) {
+  auto on_complete = base::BindOnce(
+      [](LikeMessageCallback callback, APIRequestResult result) {
+        if (result.Is2XXResponseCode() && result.value_body().is_dict()) {
+          std::string id = *result.value_body().GetDict().FindString("id");
+          std::move(callback).Run(id);
+          return;
+        }
+        std::move(callback).Run(absl::nullopt);
+      },
+      std::move(callback));
+
+  if (active_chat_tab_helper_) {
+    ai_chat::RatingPayload rating_payload;
+    rating_payload.id = 0;
+    rating_payload.message = turn_text;
+
+    feedback_api_->SendRating(is_liked, rating_payload,
+                              active_chat_tab_helper_->GetCurrentModel().name,
+                              std::move(on_complete));
+  }
+}
+
+void AIChatUIPageHandler::SendFeedback(const std::string& category,
+                                       const std::string& feedback,
+                                       const std::string& rating_id) {
+  auto on_complete = base::BindOnce([](APIRequestResult result) {
+    // TODO(nullhook): Do something
```

Comment:
For this, and `LikeMessage`, let's call a function on the f/e so that it can display the success / fail alert center message at that point.